packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,386 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/SafeCast.sol";
+
+import "./interfaces/IFederatedAttestations.sol";
+import "../common/interfaces/IAccounts.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/Signatures.sol";
+import "../common/UsingPrecompiles.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+/**
+ * @title Contract mapping identifiers to accounts
+ */
+contract FederatedAttestations is
+  IFederatedAttestations,
+  ICeloVersionedContract,
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ReentrancyGuard,
+  UsingPrecompiles
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  struct OwnershipAttestation {
+    address account;
+    uint256 issuedOn;
+    address signer;
+  }
+
+  // TODO ASv2 revisit linting issues & all solhint-disable-next-line max-line-length
+
+  // identifier -> issuer -> attestations
+  mapping(bytes32 => mapping(address => OwnershipAttestation[])) public identifierToAddresses;
+  // account -> issuer -> identifiers
+  mapping(address => mapping(address => bytes32[])) public addressToIdentifiers;
+  // signer => isRevoked
+  mapping(address => bool) public revokedSigners;
+
+  bytes32 public constant EIP712_VALIDATE_ATTESTATION_TYPEHASH = keccak256(
+    "OwnershipAttestation(bytes32 identifier,address issuer,address account,uint256 issuedOn)"
+  );
+  bytes32 public eip712DomainSeparator;
+
+  event EIP712DomainSeparatorSet(bytes32 eip712DomainSeparator);
+  event AttestationRegistered(
+    bytes32 indexed identifier,
+    address indexed issuer,
+    address indexed account,
+    uint256 issuedOn,
+    address signer
+  );
+  event AttestationDeleted(
```

Comment:
We may want to include `issuedOn` in this event since it is needed to uniquely identify an attestation in the case that it is deleted, then reissued.